### PR TITLE
Create list of runs to be deleted from the DR servers

### PR DESCRIPTION
**Pull request by jenniferliddle**

### CODE REVIEW

File: bin/npg_deletable_dr_runs [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+            'help|?'   => \$help)
+  or croak("Error in command line arguments\n");
+
+pod2usage(1) if $help;
+
+#my $rs=npg_tracking::Schema->connect()->resultset("Run")->search();
+my $rs=npg_tracking::Schema->connect()->resultset(q(RunStatusDict))
+    ->search({description=>[q[run cancelled],q[data discarded],q[qc complete]]})
+    ->related_resultset(q(run_statuses))->search({q(run_statuses.iscurrent)=>1})
+    ->related_resultset("run")->search();
```
Comment (dkj) [inline bin/npg_deletable_dr_runs]:
```suggestion
    ->related_resultset("run")->search({q(run.folder_path_glob)=>{LIKE=>q(%/esa-%)}});
```

File: bin/npg_deletable_dr_runs [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+my $total = $rs->count;
+my $run;
+my $found = 0;
+while (my $run=$rs->next()) {
+    my $status = $run->current_run_status_description || "";
+    my $tag = $run->is_tag_set("staging");
+    my $run_folder = $run->folder_name || "";
+    my $path_glob = $run->folder_path_glob || "";
+    if ($run_folder and !$tag and $path_glob=~/esa/) {
```
Comment (dkj) [inline bin/npg_deletable_dr_runs]:
```suggestion
    if ($run_folder and !$tag ) {
```
as condition checked in DBIC search above (taking query results down from 10s of thousands to hundreds)

File: bin/npg_deletable_dr_runs [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+
+pod2usage(1) if $help;
+
+#my $rs=npg_tracking::Schema->connect()->resultset("Run")->search();
+my $rs=npg_tracking::Schema->connect()->resultset(q(RunStatusDict))
+    ->search({description=>[q[run cancelled],q[data discarded],q[qc complete]]})
+    ->related_resultset(q(run_statuses))->search({q(run_statuses.iscurrent)=>1})
+    ->related_resultset("run")->search({q(run.folder_path_glob)=>{LIKE=>q(%/esa-%)}});
+my $total = $rs->count;
+my $run;
```
Comment (mgcam) [inline bin/npg_deletable_dr_runs]: not used, can be deleted

File: bin/npg_deletable_dr_runs [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+my $found = 0;
+while (my $run=$rs->next()) {
+    my $status = $run->current_run_status_description || "";
+    my $tag = $run->is_tag_set("staging");
+    my $run_folder = $run->folder_name || "";
+    my $path_glob = $run->folder_path_glob || "";
```
Comment (mgcam) [inline bin/npg_deletable_dr_runs]: not used, can be deleted

File: bin/npg_deletable_dr_runs [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+
+my $help;
+my $verbose;
+GetOptions ('verbose' => \$verbose,
+            'help|?'   => \$help)
+  or croak("Error in command line arguments\n");
+
+pod2usage(1) if $help;
+
+#my $rs=npg_tracking::Schema->connect()->resultset("Run")->search();
```
Comment (mgcam) [inline bin/npg_deletable_dr_runs]: commented out line can be deleted

File: bin/npg_deletable_dr_runs [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+if ($verbose) { print STDERR "\nFound ", $found, " out of ", $total, " records\n"; }
+
+
+1;
+
+__END__
+
+=head1 NAME
+
+deletable_dr_runs
```
Comment (mgcam) [inline bin/npg_deletable_dr_runs]: script name has changed

File: bin/npg_deletable_dr_runs [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+
+=head1 USAGE
+
+  deletable_dr_runs
+  deletable_dr_runs --help
+
+=head1 SYNOPSIS
+
+  Script to read the tracking database and output a list of
+  runfolder names that can be deleted from the Disaster Recovery hxi
```
Comment (mgcam) [inline bin/npg_deletable_dr_runs]: not sure what hxi is

File: bin/npg_deletable_dr_runs [hunk 8]
```diff
@@ -0,0 +1,126 @@
...
+    my $tag = $run->is_tag_set("staging");
+    my $run_folder = $run->folder_name || "";
+    my $path_glob = $run->folder_path_glob || "";
+    if ($run_folder and !$tag ) {
+        print $run_folder, "\t", $run->id_run, "\t", $run->folder_path_glob, "\t", $status, "\n";
+        $found++;
+    }
+}
+
+if ($verbose) { print STDERR "\nFound ", $found, " out of ", $total, " records\n"; }
```
Comment (mgcam) [inline bin/npg_deletable_dr_runs]: Probably the verbose option is not needed, the found number follows from the number of lines, the total is not telling anything important. If left should be added to a list of options in POD

File: bin/npg_deletable_dr_runs (hunk 6 above)
Comment (dkj) [reply bin/npg_deletable_dr_runs]: Oh yes, and in Usage

File: bin/npg_deletable_dr_runs [hunk 9]
```diff
@@ -0,0 +1,116 @@
...
+
+use npg_tracking::Schema;
+
+our $VERSION = '0';
+
+my $help;
+GetOptions ('help|?'   => \$help)
+  or croak("Error in command line arguments\n");
+
+pod2usage(1) if $help;
```
Comment (mgcam) [inline bin/npg_deletable_dr_runs]: pod2usage(0) if $help; # not an error since --help is a legitimate option

**Comment by mgcam** [comment]
@jenniferliddle, thank you for the changes

**Pull request merged**